subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/StartParameterResolutionOverride.java

```diff
@@ -79,12 +81,19 @@ public DependencyVerificationOverride dependencyVerificationOverride(BuildOperat
         List<String> checksums = startParameter.getWriteDependencyVerifications();
         if (!checksums.isEmpty()) {
             SingleMessageLogger.incubatingFeatureUsed("Dependency verification");
-            return new WriteDependencyVerificationFile(currentDir, buildOperationExecutor, checksums, checksumService);
+            return new WriteDependencyVerificationFile(currentDir, buildOperationExecutor, checksums, checksumService, startParameter.isDryRun());
         } else {
             File verificationsFile = DependencyVerificationOverride.dependencyVerificationsFile(currentDir);
             if (verificationsFile.exists()) {
+                if (startParameter.getDependencyVerificationMode() == DependencyVerificationMode.OFF) {
+                    return DependencyVerificationOverride.NO_VERIFICATION;
+                }
                 SingleMessageLogger.incubatingFeatureUsed("Dependency verification");
-                return new ChecksumVerificationOverride(buildOperationExecutor, verificationsFile, checksumService);
+                try {
+                    return new ChecksumVerificationOverride(buildOperationExecutor, verificationsFile, checksumService, startParameter.getDependencyVerificationMode());
+                } catch (Exception e) {
+                    return new FailureVerificationOverride(e, verificationsFile);
```

Comment:
I am not sure I understand the real benefit of this change.
It effectively enables to run `help` or `tasks` but not much more.
Is there something else I am missing for this feature?

Reply:
It's more than this. Without this change, Gradle fails to initialize: the `DependencyManagementBuildScopeServices` fail to create their dependent services, so the error that is thrown to the user is not nice. With this change, the error is captured and thrown _after_ initialization.

Reply:
Ah ok, makes sense then! 👍